python/lsst/meas/extensions/multiprofit/fit_coadd_multiband.py

```diff
@@ -19,54 +19,106 @@
+        dtype=str,
+        default=[],
+        doc="list of bandpass filters to fit",
+        listCheck=lambda x: len(set(x)) == len(x),
+    )
+    mask_names_zero = pexConfig.ListField[str](
+        doc="Mask bits to mask out",
+        default=["BAD", "EDGE", "SAT", "NO_DATA"],
+    )
+    psf_sigma_subtract = pexConfig.Field[float](
+        doc="Default x/y sigma value to subtract in quadrature from best-fit PSF sigma values",
+        default=0.1,
+        check=lambda x: np.isfinite(x) and (x > 0),
+    )
+    prefix_column = pexConfig.Field[str](default="mpf_", doc="Column name prefix")
+
+    def bands_read_only(self) -> set[str]:
+        # TODO: Re-implement determination of prior-only bands once
+        # data-driven priors are re-implemented (DM-4xxxx)
+        return set()
+
+    def setDefaults(self):
+        super().setDefaults()
+        self.flag_errors = {
+            IsParentError.column_name(): "IsParentError",
+            NotPrimaryError.column_name(): "NotPrimaryError",
+            PsfRebuildFitFlagError.column_name(): "PsfRebuildFitFlagError",
+        }
+        self.centroid_pixel_offset = -0.5
 
 
 @dataclass(frozen=True, kw_only=True, config=fitMB.CatalogExposureConfig)
 class CatalogExposurePsfs(fitMB.CatalogExposureInputs, CatalogExposureSourcesABC):
     """Input data from lsst pipelines, parsed for MultiProFit."""
 
     channel: g2f.Channel = pydantic.Field(title="Channel for the image's band")
-    config_fit: CatalogSourceFitterConfig = pydantic.Field(title="Channel for the image's band")
+    config_fit: MultiProFitSourceConfig = pydantic.Field(title="Config for fitting options")
 
     def get_psfmodel(self, source):
         match = np.argwhere(
-            (self.table_psf_fits[self.config_fit_psf.column_id] == source[self.config_fit.column_id])
+            (self.table_psf_fits[self.psfmodel_data.config.column_id] == source[self.config_fit.column_id])
         )[0][0]
-        return self.config_fit_psf.rebuild_psfmodel(self.table_psf_fits[match])
-
-    def get_source_observation(self, source) -> g2f.Observation:
-        if (not source["detect_isPrimary"]) or source["merge_peak_sky"]:
-            raise NotPrimaryError(f"source {source[self.config_fit.column_id]} has invalid flags for fit")
+        psfmodel = self.psfmodel_data.psfmodel
+        self.psfmodel_data.init_psfmodel(self.table_psf_fits[match])
+
+        sigma_subtract = self.config_fit.psf_sigma_subtract
+        if sigma_subtract > 0:
+            sigma_subtract_sq = sigma_subtract * sigma_subtract
+            for param in self.psfmodel_data.parameters.values():
+                if (
+                    isinstance(param, g2f.SigmaXParameterD)
+                    or isinstance(param, g2f.SigmaYParameterD)
+                    or isinstance(param, g2f.ReffXParameterD)
+                    or isinstance(param, g2f.ReffYParameterD)
+                ):
+                    param.value = math.sqrt(param.value**2 - sigma_subtract_sq)
```

Comment:
You don't do anything sigma_subtract is negative. Is this normal/expected, or should something be added here to either throw an exception or update something?

Reply:
There's already a check on the field that it's finite and >0 (which I will change to be >=0, as it should have been to start with).